indico/htdocs/js/indico/modules/category_management.js

```diff
@@ -0,0 +1,92 @@
+/* This file is part of Indico.
+ * Copyright (C) 2002 - 2016 European Organization for Nuclear Research (CERN).
+ *
+ * Indico is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Indico is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Indico; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+(function() {
+    'use strict';
+
+    $(document).ready(function() {
+        setupCategoriesTable();
+    });
+
+    function setupCategoriesTable() {
+        var $table = $('table.categories-management');
+        var $tbody = $table.find('tbody');
+        var $categoryRowSelector = 'tr[data-category-id]';
+
+        $('.js-sort-categories').on('click', function() {
+            var currentOrder = getCategoriesOrder();
+            function undo() {
+                restoreCategoriesOrder(currentOrder);
+                return setOrderAjax(currentOrder);
+            }
+            sortCategories();
+            setOrderAjax(getCategoriesOrder());
+            var message = cornerMessage({
+                actionLabel: $T.gettext("Undo"),
+                actionCallback: undo,
+                feedbackMessage: $T.gettext("The category list order has been restored."),
+                duration: 10000,
+                message: $T.gettext("The category list has been sorted.")
+            });
+        });
+
+        $tbody.sortable({
+            axis: 'y',
+            containment: 'parent',
+            cursor: 'move',
+            handle: '.js-handle',
+            items: '> tr',
+            update: function() {
+                setOrderAjax(getCategoriesOrder());
+            }
+        });
+
+        function getCategoriesOrder() {
+            return $.map($tbody.find($categoryRowSelector), function(row) {
+                return $(row).data('category-id');
+            });
+        }
+
+        function restoreCategoriesOrder(order) {
+            $.each(order, function(index, id) {
+                $tbody.find('[data-category-id=' + id + ']').detach().appendTo($tbody);
+            });
+        }
+
+        function sortCategories() {
+            var sortedRows = $tbody.find($categoryRowSelector).sort(function(a, b) {
```

Comment:
you can simplify this a bit:

    $tbody.find($categoryRowSelector).detach().sort(...).appendTo($tbody);